controllers/apidefinition_controller.go

```diff
@@ -159,6 +176,72 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			}
 		}
 
+		if upstreamRequestStruct.Spec.GraphQL != nil && upstreamRequestStruct.Spec.GraphQL.GraphRef != "" {
+			if upstreamRequestStruct.Spec.GraphQL.ExecutionMode == model.SubGraphExecutionMode {
+				subgraph := &tykv1alpha1.SubGraph{}
+
+				err := r.Client.Get(ctx, types.NamespacedName{
+					Namespace: req.Namespace,
+					Name:      desired.Spec.GraphQL.GraphRef,
+				}, subgraph)
+				if err != nil {
+					return err
+				}
+
+				upstreamRequestStruct.Spec.GraphQL.Schema = subgraph.Spec.Schema
+				upstreamRequestStruct.Spec.GraphQL.Subgraph.SDL = subgraph.Spec.SDL
+
+				subgraph.Status.APIID = upstreamRequestStruct.Spec.APIID
```

Comment:
During the call we decided that we shouldn't allow linking of multiple API definition to subgraph. So if Status of subgraph already contains APIID, we should throw error.

Reply:
Also when an APIDefinition is deleted we should clear this field.

Reply:
I see a problem here. How we will remove link of subgraph if APIDefinition is updated and graph_ref field is set to ""?

Reply:
I was thinking of making addition of Subgraph ID to ApiDefinition.Status. So that, instead of the code block that you commented on GitHub, we can follow following structure.
```
if execution_mode == subgraph:
	if graph_ref == "":
		subgraph = r.Client.Get(desired.status.subgraph)
		subgraph.status.APIID = ""
	else:
		if subgraph.status.APIID != desired.spec.APIID:
			throw error ("cannot link one subgraoh to multiple api def"
		...
		rest of the operation is same
```
Also, for the deletion, we can get subgraph by looking at ApiDefinition’s Status and implement following mechanism:
```
subgraph = r.Client.Get(desired.status.subgraph)
subgraph.status.APIID == ""
```

Let's handle this part in separate PR, since this PR is already big enough to review.